### ISSUE TITLE
Cannot view blog posts: 'NoneType' object has no attribute 'paginate_by'

### ISSUE THREAD

**Issue by sedrubal** [nephila/djangocms-blog#184]
When I add some blog posts I always get errors... This time it's this error:

```
AttributeError at /de/

'NoneType' object has no attribute 'paginate_by'

Request Method: 	GET
Request URL: 	http://localhost:8000/de/
Django Version: 	1.8.7
Exception Type: 	AttributeError
Exception Value: 	

'NoneType' object has no attribute 'paginate_by'

Exception Location: 	/venv/lib/python3.4/site-packages/djangocms_blog/views.py in get_paginate_by, line 66
```

### OS:

`Fedora:23`

### Details about local setup:

It's [this project](https://github.com/fau-fablab/website/tree/djangocms_blog). If you want you can inspect all files and even run the docker container...

- django `1.8.7`
- djangocms `3.2.0`
- djangocms blog `0.3.6`
- packages from pypi
- python 3
- virtual environment

### Steps to reproduce:

It's hard to describe as I have done this steps some times and everything an other error occurred.

- I installed djangocms as described in the docs (I think)
- I moved the blog to the top in the admins view so that it appears under `http://mysite.com/`
- I added two blog posts in admin view
- I navigated to `/` and got the error

Merry christmas ;) :christmas_tree: :santa:

**Comment by yakky** [comment]
Apparently the apphook config is not loaded. I'll check the behavior on the root
Are you really using ``djangocms blog=0.3.6``?

**Comment by sedrubal** [comment]
Yes, `pip freeze` says: `djangocms-blog==0.6.3`, thanks

**Comment by yakky** [comment]
you have to reload the project every time you move a page with an apphook associated (this is a django CMS requirement, not blog one)

**Comment by yakky** [comment]
Closing this to due lack of feedback
@sedrubal   feel free to reopen if you have more details

**Comment by sedrubal** [comment]
Sorry for not answering...

What do you mean with reload? Restart the webserver? This does not help...

(I can't reopen this issue ;) )

**Comment by ondrejsika** [comment]
I have a same problem

**Comment by yakky** [comment]
@sedrubal @ondrejsika do you have at least one model in /admin/djangocms_blog/blogconfig/?
When you added the apphook to the page, did you selected said configuration under "
Application configurazions" in the page advanced settings?

**Comment by ondrejsika** [comment]
Yes, but `aldryn_apphook_config.utils.get_app_instance` returns no config (`None`)

**Comment by yakky** [comment]
@ondrejsika which django version are you on?

**Comment by ondrejsika** [comment]
We are using django 1.6.11. I think the problem may be there, I used this:

```
...
def get_paginate_by(self, queryset):
    return getattr(self.config, 'paginate_by', None) or get_setting('PAGINATION')
...
```

instead of this:

```
...
def get_paginate_by(self, queryset):
    return self.config.paginate_by or get_setting('PAGINATION')
...
```

at the `views.py` and it probably worked. But I didn't see any posts :(

**Comment by yakky** [comment]
We can easily change ``self.config.paginate_by`` to ``getattr(self.config, 'paginate_by', None)``, but if ``self.config`` is None there are underlying issues that needs to be fixed (that's the reason why ``self.config`` is assumed to exist in ``get_paginate_by`` and elsewhere)
Are you available for some debugging (eventually on irc) to dig a little deeper?
Apparently aldryn-apphooks-config can't retrieve the configuration (function is https://github.com/aldryn/aldryn-apphooks-config/blob/master/aldryn_apphooks_config/utils.py#L19-L31)
Which django CMS version are you on?

**Comment by ondrejsika** [comment]
I dont have too much time to deeper debugging, sorry :( but you can contact me directly by email <ondrej@ondrejsika.com> or by jabber (same as email).

We are using `django-cms==3.1.2`. This method returns `None` https://github.com/aldryn/aldryn-apphooks-config/blob/master/aldryn_apphooks_config/utils.py#L27. When i tried to force return app.app_config, there is no attribute `paginate_by`

**Comment by yakky** [comment]
@ondrejsika the interesting data would be:
- ``app`` value @ https://github.com/aldryn/aldryn-apphooks-config/blob/master/aldryn_apphooks_config/utils.py#L21
- ``namespace`` value @ https://github.com/aldryn/aldryn-apphooks-config/blob/master/aldryn_apphooks_config/utils.py#L26
- ``request.path_info`` and ``request.path`` value @ https://github.com/aldryn/aldryn-apphooks-config/blob/master/aldryn_apphooks_config/utils.py#L26

This is related to CMS / apphooks config, rather than blog, anyway.
I'd say to update the CMS to 3.1.5, but I don't remember of any major change in the apphook area

**Comment by ondrejsika** [comment]
there are values:

```
app <djangocms_blog.cms_app.BlogApp object at 0x7f7f51bfa510>
app.app_config <class 'djangocms_blog.cms_appconfig.BlogConfig'>
request <WSGIRequest ... >
request.path /blog/2016/02/09/hello-miners/
namespace djangocms_blog
```

generated by

```
    app = None
    if getattr(request, 'current_page', None) and request.current_page.application_urls:
        app = apphook_pool.get_apphook(request.current_page.application_urls)

    if app and app.app_config:
        print('app', app)
        print('app.app_config', app.app_config)
        try:
            config = None
            with override(get_language_from_request(request, check_path=True)):
                print('request', request)
                print('request.path', request.path)
                namespace = resolve(request.path).namespace
                print('namespace', namespace)
                config = app.get_config(namespace)
                print('config', namespace)
            return namespace, config
        except Resolver404:
            pass
    return '', None
```

**Comment by yakky** [comment]
Does the apphook have all the translations? I can't see the output of ``print('config', namespace)``

**Comment by ondrejsika** [comment]
Sorry, there is `None` and the correct from may be `print('config', config)`.

I've tested only with English.

**Comment by ondrejsika** [comment]
I have this in blog config tables.

```
mysql> select * from djangocms_blog_blogconfig;
+----+-----------------------------------------+-----------+----------+
| id | type                                    | namespace | app_data |
+----+-----------------------------------------+-----------+----------+
|  1 | djangocms_blog.cms_appconfig.BlogConfig | Blog      | {}       |
+----+-----------------------------------------+-----------+----------+
1 row in set (0.00 sec)

mysql> select * from djangocms_blog_blogconfig_translation;
+----+---------------+-----------+-----------+-------------+
| id | language_code | app_title | master_id | object_name |
+----+---------------+-----------+-----------+-------------+
|  1 | en            | Blog      |         1 | Article     |
|  2 | de            | Blog      |         1 | Article     |
|  3 | fr            | Blog      |         1 | Article     |
|  4 | ru            | Blog      |         1 | Article     |
|  5 | cs            | Blog      |         1 | Article     |
|  6 | pl            | Blog      |         1 | Article     |
|  7 | nl            | Blog      |         1 | Article     |
|  8 | es            | Blog      |         1 | Article     |
|  9 | pt-br         | Blog      |         1 | Article     |
| 10 | zh            | Blog      |         1 | Article     |
| 11 | ro            | Blog      |         1 | Article     |
| 12 | hu            | Blog      |         1 | Article     |
| 13 | it            | Blog      |         1 | Article     |
| 14 | tr            | Blog      |         1 | Article     |
+----+---------------+-----------+-----------+-------------+
14 rows in set (0.00 sec)

```

**Comment by yakky** [comment]
@ondrejsika thanks for the great information, namespace for djangocms_blog_blogconfig does not match the one that come from the request, thus I guess there is something wrong in the CMS / apphook config.
How about the content of the  ``cms_page`` table for the page to which the blog is attached to (``select * from cms_page where application_urls='BlogApp'``)?
In the meanwhile, by changing the namespace to ``djangocms_blog`` everything should work

**Comment by yakky** [comment]
@ondrejsika did you change the "instance namespace" value in ``/admin/djangocms_blog/blogconfig/2/`` after adding the apphook to the page?

**Comment by quinox** [comment]
I can trigger the same error messages as follows:

* Make a bilingual page with distinct slugs, such as `news` for EN and `nieuws` for NL
* Hook the page up to the blog application
* Open `/en/news/`, it works: it renders all English blog posts
* Open `/nl/nieuws/`, it works: it renders all Dutch blog posts
* Open `/nl/news/`, it crashes with the paginate_by error
* Open `/en/nieuws/`, it crashes with the paginate_by error

Tested it with:
* django-cms==3.1.2 / djangocms-blog==0.6.3
* django-cms==3.1.5 / djangocms-blog==0.7.0b1

**Comment by yakky** [comment]
@quinox interesting case. I'll try to reproduce. It looks more an issue with aldryn-apphooks-config than the blog itself.
I hope to provide more feedback soon

**Comment by yakky** [comment]
@quinox @sedrubal @ondrejsika aldryn-apphooks-config 0.2.7 should fix some of these issues
I opened a issue there to track it. It has little to do with the blog, although I'll improve how the blog can mitigate these issues

**Comment by yakky** [comment]
You can test linked PR

**Comment by yakky** [comment]
@quinox @sedrubal @ondrejsika  I released djangocms_blog 0.7.0b2 which includes the mitigation fix

**Comment by quinox** [comment]
Thanks, 0.7.0b2 no longer crashes for me.

**Comment by yakky** [comment]
@quinox thanks for the feedback

**Comment by yakky** [comment]
Fixed in 0.7.0b2
